pennylane/interfaces/autograd.py

```diff
@@ -97,10 +97,12 @@ def grad_fn(dy):
 logger.addHandler(logging.NullHandler())
 
 
+# pylint: disable=unused-argument
 def autograd_execute(
     tapes: Batch,
     execute_fn: ExecuteFn,
     jpc: qml.interfaces.jacobian_products.JacobianProductCalculator,
+    device=None,

```

Comment:
I do think it could be feasible to remove this in the future if we either:

A) No longer need `device` to determine the shape
B) Directly wrap `execute_fn` and `JacobianProductCalculator` before even sending it to the interface code.

But this is a step in the correct direction and gets the required features.

Reply:
And I'll probably have to add an `is_differentiable` keyword argument for tensorflow 😢 though I'm still trying to figure out if there's a better way.